sklearn/utils/estimator_checks.py

```diff
@@ -1460,6 +1472,69 @@ def check_estimators_dtypes(name, estimator_orig, strict_mode=True):
                 getattr(estimator, method)(X_train)
 
 
+def check_transformer_preserve_dtypes(
+    name, transformer_orig, strict_mode=True
+):
+    # check that dtype are preserved meaning if input X is of some dtype
+    # X_transformed should be from the same dtype.
+    X, y = make_blobs(
+        n_samples=30,
+        centers=[[0, 0, 0], [1, 1, 1]],
+        random_state=0,
+        cluster_std=0.1,
+    )
+    X = StandardScaler().fit_transform(X)
+    X -= X.min()
+    X = _pairwise_estimator_convert_X(X, transformer_orig)
+
+    def _fit_and_transform(transformer, X, y):
+        transformer = clone(transformer)
+        set_random_state(transformer)
+        X_trans = transformer.fit_transform(X, y)
+
+        if isinstance(X_trans, tuple):
+            # cross-decompostion returns a tuple of (x_scores, y_scores)
+            # when given y with fit_transform; only check the first element
+            X_trans = X_trans[0]
+        return X_trans
+
+    tags_transformer = transformer_orig._get_tags()
+    transformer_default_dtype = tags_transformer["preserves_dtype"][0]
```

Comment:
It seems that the list (`tags_transformer["preserves_dtype"]`) can be empty.

Shouldn't this fail for such transformers? Is this test not run for them?

Reply:
The test will be skipped because `if tags["preserves_dtype"]` in the `yield_*` function will be False.
This was suggested by @thomasjpfan in some early comment.